### PR TITLE
PubMatic 1.0 adapter

### PR DESCRIPTION
**Pull request by PubMatic-OpenWrap**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'pubmatic',
  params: {
	publisherId: '301',                  // required
	adSlot: '/15671365/DMDemo@728x90',   // required
	pmzoneid: 'zone1, zone11',           // optional
	lat: '40.712775',                    // optional
	lon: '-74.005973',                   // optional
	yob: '1982',                         // optional
	kadpageurl: 'www.test.com',          // optional							      
	gender: 'M',                         // optional
	kadfloor: '0.50'                     // optional 									      
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: header-bidding@pubmatic.com
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by mkendall07** [comment]
@PubMatic-OpenWrap 
@mike-chowla 
Thanks for updating to 1.0! Can you please update the documentation site for your bidder, I see lots of params that are not documented here: http://prebid.org/dev-docs/bidders.html#pubmatic
source: https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  if (!conf.kadpageurl) {
+    conf.kadpageurl = conf.pageURL;
+  }
+
+  var key, value, entry;
+  for (key in CUSTOM_PARAMS) {
+    // istanbul ignore else
+    if (CUSTOM_PARAMS.hasOwnProperty(key)) {
+      value = params[key];
+      // istanbul ignore else
```
Comment (jsnellbaker) [inline modules/pubmaticBidAdapter.js]: Are these types of comments still needed?  I assume they were part of the development process and wanted to confirm.

File: modules/pubmaticBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        return false;
+      }
+      if (!utils.isStr(bid.params.adSlot)) {
+        utils.logWarn('PubMatic: adSlotId is mandatory and cannot be numeric. Call to OpenBid will not be sent.');
+        return false;
+      }
+      return true;
+    }
+    return false;
+    // return !!(bid && bid.params && utils.isStr(bid.params.publisherId) && utils.isStr(bid.params.adSlot));
```
Comment (jsnellbaker) [inline modules/pubmaticBidAdapter.js]: Can we remove this comment if the earlier code is the planned/definite approach to perform the validation check?

**Comment by PubMatic-OpenWrap** [comment]
We have removed unwanted comments.
Also updated .md file with userSync settings and new test ids.

**Comment by jsnellbaker** [comment]
@PubMatic-OpenWrap Thanks for making the updates.  I used the updated test credentials and I saw the test ad deliver fine.  LGTM

File: modules/pubmaticBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        } else {
+          utils.logWarn('PubMatic: Ignoring param : ' + key + ' with value : ' + CUSTOM_PARAMS[key] + ', expects string-value, found ' + typeof value);
+        }
+      }
+    }
+  }
+  return conf;
+}
+
+function _createOrtbTemplate(conf) {
```
Comment (mkendall07) [inline modules/pubmaticBidAdapter.js]: nice! Would be great if we had a community ORTB adapter at some point...

**Pull request merged**